core/ipc/rpc/utils/src/main/java/org/opennms/core/rpc/utils/mate/EntityScopeProviderImpl.java

```diff
@@ -87,11 +97,100 @@ public Scope getScopeForNode(final Integer nodeId) {
                             .map("node", "sys-location", (n) -> Optional.ofNullable(n.getSysLocation()))
                             .map("node", "sys-contact", (n) -> Optional.ofNullable(n.getSysContact()))
                             .map("node", "sys-description", (n) -> Optional.ofNullable(n.getSysDescription()))
+                            .map("node", "sys-object-id", (n) -> Optional.ofNullable(n.getSysObjectId()))
                             .map("node", "location", (n) -> Optional.ofNullable(n.getLocation().getLocationName()))
-                            .map("node", "area", (n) -> Optional.ofNullable(n.getLocation().getMonitoringArea()))
-            );
+                            .map("node", "area", (n) -> Optional.ofNullable(n.getLocation().getMonitoringArea())));
         });
+        return metaDataScope;
+    }
 
+    private Optional<String> getNodeCriteria(final OnmsNode node) {
+        Objects.requireNonNull(node, "Node can not be null");
+        if (node.getForeignSource() != null) {
+            return Optional.of(node.getForeignSource() + ":" + node.getForeignId());
+        } else {
+            return Optional.of(Integer.toString(node.getId()));
+        }
+    }
+
+    @Override
+    public Scope getScopeForAssets(final Integer nodeId) {
```

Comment:
I have mixed feelings about having a new method. If we can merge this with `getScopeForNode` by just adding the `Scope` created here to the Nodes `FallbackScope`, all existing usages would benefit from the available data.

Reply:
I'm easy here. I merged the 2 methods.